x/fees/keeper/fees.go

```diff
@@ -0,0 +1,178 @@
+package keeper
+
+import (
+	"github.com/cosmos/cosmos-sdk/store/prefix"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/ethereum/go-ethereum/common"
+
+	"github.com/tharsis/evmos/v3/x/fees/types"
+)
+
+// GetAllFees - get all registered DevFeeInfo instances
+func (k Keeper) GetAllFees(ctx sdk.Context) []types.DevFeeInfo {
+	fees := []types.DevFeeInfo{}
+
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixFee)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var fee types.DevFeeInfo
+		k.cdc.MustUnmarshal(iterator.Value(), &fee)
+
+		fees = append(fees, fee)
+	}
+
+	return fees
+}
+
+// IterateFees iterates over all registered `DevFeeInfos` and performs a
+// callback.
+func (k Keeper) IterateFees(
+	ctx sdk.Context,
+	handlerFn func(fee types.DevFeeInfo) (stop bool),
+) {
+	store := ctx.KVStore(k.storeKey)
+	iterator := sdk.KVStorePrefixIterator(store, types.KeyPrefixFee)
+	defer iterator.Close()
+
+	for ; iterator.Valid(); iterator.Next() {
+		var fee types.DevFeeInfo
+		k.cdc.MustUnmarshal(iterator.Value(), &fee)
+
+		if handlerFn(fee) {
+			break
+		}
+	}
+}
+
+// GetFeeInfo returns DevFeeInfo for a registered contract
+func (k Keeper) GetFeeInfo(ctx sdk.Context, contract common.Address) (types.DevFeeInfo, bool) {
+	deployerAddress, found := k.GetDeployer(ctx, contract)
+	if !found {
+		return types.DevFeeInfo{}, false
+	}
+	withdrawalAddress, _ := k.GetWithdrawal(ctx, contract)
+	feeInfo := types.DevFeeInfo{
+		ContractAddress: contract.String(),
+		DeployerAddress: deployerAddress.String(),
+		WithdrawAddress: withdrawalAddress.String(),
+	}
+	return feeInfo, true
+}
+
+// GetDeployer returns the deployer address for a registered contract
+func (k Keeper) GetDeployer(ctx sdk.Context, contract common.Address) (sdk.AccAddress, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixFee)
+	bz := store.Get(contract.Bytes())
+	if len(bz) == 0 {
+		return sdk.AccAddress{}, false
+	}
+	return sdk.AccAddress(bz), true
+}
+
+// GetWithdrawal returns the withdrawal address for a registered contract
+func (k Keeper) GetWithdrawal(ctx sdk.Context, contract common.Address) (sdk.AccAddress, bool) {
+	store := prefix.NewStore(ctx.KVStore(k.storeKey), types.KeyPrefixFeeWithdrawal)
+	bz := store.Get(contract.Bytes())
+	if len(bz) == 0 {
+		return k.GetDeployer(ctx, contract)
+	}
+	return sdk.AccAddress(bz), true
+}
+
+// SetFee stores a registered contract
+func (k Keeper) SetFee(ctx sdk.Context, contract common.Address, deployer sdk.AccAddress, withdrawal *sdk.AccAddress) {
```

Comment:
you can set the address as nil already, no need for a pointer

```suggestion
func (k Keeper) SetFee(ctx sdk.Context, contract common.Address, deployer, withdrawal sdk.AccAddress) {
```